src/platform/plugins/shared/controls/public/actions/delete_control_action.tsx

```diff
@@ -78,6 +79,7 @@ export class DeleteControlAction implements Action<EmbeddableApiContext> {
       .then((confirmed) => {
         if (confirmed) {
           embeddable.parentApi.removePanel(embeddable.uuid);
+          if (isVariablesControl(embeddable)) embeddable.clearVariables();
```

Comment:
How about moving this logic into `src/platform/plugins/shared/controls/public/control_group/init_controls_manager.ts` `removePanel` implementation. You can access the embeddable api with `getControlApi` method in that same file. That way, any time parentApi.removePanel is called, variables are cleared. 

This will prevent bugs where controls are deleted via other mechanisms, for example, on the controls settings flyout there is a "delete all controls" button

Reply:
One thing this is causing right now is that `reset` doesn't update the actual variable values. It will properly reset the selection of the Control, but since what is selected on the control isn't actually the source of truth for the variables service, the ESQL charts will not update.

![var reset](https://github.com/user-attachments/assets/980d087e-243b-48da-8ed0-d0ac7beed55f)

Reply:
Hmmm good catch, can you help me fix it? I don't know how this logic work